backend/internal/handlers/utils.go

```diff
@@ -0,0 +1,81 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/golang/gddo/httputil/header"
+)
+
+type malformedRequest struct {
+	status int
+	msg    string
+}
+
+func (mr *malformedRequest) Error() string {
+	return mr.msg
+}
+
+// From https://www.alexedwards.net/blog/how-to-properly-parse-a-json-request-body
+func decodeJSONBody(w http.ResponseWriter, r *http.Request, dst interface{}) error {
+	if r.Header.Get("Content-Type") != "" {
+		value, _ := header.ParseValueAndParams(r.Header, "Content-Type")
```

Comment:
you're vendoring the entire godoc.org implementation for a relatively simple function.
you can replace it with `if !strings.HasPrefix(r.Header.Get("Content-Type", "application/json"))` and drop a large dependency